src/SFA.DAS.ProviderUrlHelper/LinkGenerator.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using SFA.DAS.AutoConfiguration;
+
+namespace SFA.DAS.ProviderUrlHelper
+{
+    public class LinkGenerator : ILinkGenerator
+    {
+        private readonly Lazy<ProviderUrlConfiguration> _lazyProviderConfiguration;
+
+        public LinkGenerator(IAutoConfigurationService autoConfigurationService)
+        {
+            _lazyProviderConfiguration =
+                new Lazy<ProviderUrlConfiguration>(() => LoadProviderUrlConfiguration(autoConfigurationService));
+        }
+
+        public string ProviderCommitmentsLink(string path)
+        {
+            var configuration = _lazyProviderConfiguration.Value;
+            var baseUrl = configuration.ProviderCommitmentsBaseUrl;
+
+            return Action(baseUrl, path);
+        }
+
+        public string ProviderApprenticeshipServiceLink(string path)
+        {
+            var configuration = _lazyProviderConfiguration.Value;
+            var baseUrl = configuration.ProviderApprenticeshipServiceBaseUrl;
+
+            return Action(baseUrl, path);
+        }
+
+        public string ReservationsLink(string path)
+        {
+            var configuration = _lazyProviderConfiguration.Value;
+            var baseUrl = configuration.ReservationsBaseUrl;
+
+            return Action(baseUrl, path);
+        }
+
```

Comment:
The 3 actions above are identical apart from the Configuration Property. Would it be worthwhile writing a private method which accepts the property and path variables and returns the action link?